### ISSUE TITLE
Relay <> Dev tools plugins

### ISSUE THREAD

**Issue by Kalina-Todorova** [hashgraph/hedera-json-rpc-relay#416]
Let's investigate which are the most used dev tools plugins. Once we have this list we can test one by one to verify compatibility with the relay.

**Comment by mshakeg** [comment]
@Kalina-Todorova from the [list of hardhat community plugins](https://hardhat.org/hardhat-runner/plugins) I think the most wanted would be those used in tests such as [solidity-coverage](https://www.npmjs.com/package/solidity-coverage), [hardhat-deploy](https://www.npmjs.com/package/hardhat-deploy) and [hardhat-gas-reporter](https://www.npmjs.com/package/hardhat-gas-reporter)

**Comment by natanasow** [comment]
@mshakeg after a quick investigation:
- [solidity-coverage](https://www.npmjs.com/package/solidity-coverage) - uses `ganache-cli` and we are unable to connect it to the relay endpoint
- [hardhat-deploy](https://www.npmjs.com/package/hardhat-deploy) - 100% compatibility
- [hardhat-gas-reporter](https://www.npmjs.com/package/hardhat-gas-reporter) - 100% compatibility

**Comment by mshakeg** [comment]
@natanasow I can confirm _hardhat-deploy_ works, haven't yet tried _hardhat-gas-reporter_.

Does [solidity-coverage] not also work with the hardhat node? The [hardhat docs](https://hardhat.org/hardhat-runner/docs/guides/test-contracts#measuring-code-coverage) would indicate that it does.

**Comment by natanasow** [comment]
@mshakeg Nope, [solidity-coverage](https://www.npmjs.com/package/solidity-coverage) works fine with the hardhat node. It doesn't work with `json-rpc-relay` endpoint.

**Comment by mshakeg** [comment]
@natanasow hmm, do you know why it works on both the hardhat and ganache nodes? I'd imagine it has something to do with certain debugging methods not being supported by the hedera relay such as `debug_traceTransaction`.

**Comment by natanasow** [comment]
@mshakeg Didn't go deep, but I'm pretty sure that such debugging things as `debug_traceTransaction`, ``console.log` etc. are causing the non-compatibility.

**Comment by mshakeg** [comment]
@natanasow ok, well #477 lists non-standard relay methods some of which are relevant to this.

I don't think `console.log` is the issue as in my understanding that compiles down to a custom event emission that hardhat listens for.

**Comment by Nana-EC** [comment]
Relay supports quite a few plugins.
Additionally there's support for `debug_traceTransaction` in local node environments.
Solidity-coverage has limitations due to system contract support, that is documented

Other plugins should see separate tickets open
